src/main/scala/is/hail/methods/KinshipMatrix.scala

```diff
@@ -49,6 +51,90 @@ class KinshipMatrix(val hc: HailContext, val matrix: IndexedRowMatrix, val sampl
       .writeTable(output, hc.tmpDir, Some(sampleIds.mkString("\t")))
   }
 
+  def exportRel(output: String) {
+    prepareMatrixForExport(matrix).rows
+      .map { (row: IndexedRow) =>
+        val i = row.index
+        val arr = row.vector.toArray
+        val sb = new StringBuilder
+        var j = 0
+        while (j <= i) {
+          if (j > 0)
+            sb += '\t'
+          sb.append(arr(j))
+          j += 1
+        }
+        sb.toString()
+      }
+      .writeTable(output, hc.tmpDir)
+  }
+
+  def exportGctaGrm(output: String) {
+    val nVars = numVariantsUsed //required to avoid serialization error.
+    prepareMatrixForExport(matrix).rows
+      .map { (row: IndexedRow) =>
+        val i = row.index
+        val arr = row.vector.toArray
+        val sb = new StringBuilder
+        var j = 0
+        while (j <= i) {
+          sb.append(s"${ i + 1 }\t${ j + 1 }\t$nVars\t${ arr(j) }")
+          if (j < i) {
+            sb.append("\n")
+          }
+          j += 1
+        }
+        sb.toString()
+      }
+      .writeTable(output, hc.tmpDir)
+  }
+
+  def exportGctaGrmBin(output: String, nFile: Option[String] = None) {
+    prepareMatrixForExport(matrix).rows
+      .map { (row: IndexedRow) =>
+        val i = row.index
+        val arr = row.vector.toArray
+        val result = new Array[Byte](4 * i.toInt + 4)
+        var j = 0
+        while (j <= i) {
+          val bits: Int = java.lang.Float.floatToRawIntBits(arr(j).toFloat)
+          result(j * 4) = (bits & 0xff).toByte
+          result(j * 4 + 1) = ((bits >> 8) & 0xff).toByte
+          result(j * 4 + 2) = ((bits >> 16) & 0xff).toByte
+          result(j * 4 + 3) = (bits >> 24).toByte
+          j += 1
+        }
+        result
+      }
+      .saveFromByteArrays(output, hc.tmpDir)
+
+    nFile.foreach { f =>
+      hc.sc.hadoopConfiguration.writeDataFile(f) { s =>
+        for (_ <- 0 until sampleIds.length * (sampleIds.length + 1) / 2)
+          writeFloatLittleEndian(s, numVariantsUsed.toFloat)
+      }
+    }
+  }
+
+  def exportIdFile(idFile: String) {
+    hc.sc.hadoopConfiguration.writeTextFile(idFile) { s =>
+      for (id <- this.sampleIds) {
+        s.write(id)
+        s.write("\t")
+        s.write(id)
+        s.write("\n")
+      }
+    }
+  }
+
+  private def writeFloatLittleEndian(s: DataOutputStream, f: Float) {
+    val bits: Int = java.lang.Float.floatToRawIntBits(f)
+    s.write(bits & 0xff)
```

Comment:
It looks like it's intended to be an int based on the 8, 16, and 24 bit shifts, though I'll admit my bitshift knowledge leaves something to be desired. GRMSuite has tests involving this method., and the bin file method that uses is gets tested against PLINK